deltachat-ffi/src/lib.rs

```diff
@@ -478,15 +481,13 @@ pub unsafe extern "C" fn dc_get_chat_media(
     assert!(!context.is_null());
     let context = &*context;
 
-    if let (Some(msg_type), Some(or_msg_type2), Some(or_msg_type3)) = (
-        from_prim(msg_type),
-        from_prim(or_msg_type2),
-        from_prim(or_msg_type3),
-    ) {
-        dc_chat::dc_get_chat_media(context, chat_id, msg_type, or_msg_type2, or_msg_type3)
-    } else {
-        ptr::null_mut()
-    }
+    let msg_type = from_prim(msg_type).expect(&format!("invalid msg_type = {}", msg_type));
+    let or_msg_type2 =
+        from_prim(or_msg_type2).expect(&format!("incorrect or_msg_type2 = {}", or_msg_type2));
+    let or_msg_type3 =
+        from_prim(or_msg_type3).expect(&format!("incorrect or_msg_type3 = {}", or_msg_type3));
```

Comment:
These expect calls turn what used to be handled gracefully into a crash.  Is this the desired scenario?  Should the C API be this strict?

Reply:
Yes, this change is by intention. This `expect` is about incorrect library usage. Pre-condition violation is exactly what `assert` for.